### ISSUE TITLE
Get It Button Missing, How To Add?

### ISSUE THREAD

**Issue by jared-leddy** [SharePoint/sp-dev-docs#8324]
[Enter feedback here]

In this section below, all of our tenants/sites have this "get it" button missing. There is no reference as to how to get that button to show. Is it a feature that needs to be enabled? Is it a permission thing? How do you get the button to show when it's missing? 

Section: [https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/provision-sp-assets-from-package#update-an-existing-instance-in-the-site](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/provision-sp-assets-from-package#update-an-existing-instance-in-the-site)

Image for Ref:

![image](https://user-images.githubusercontent.com/57743183/180279782-8bd8abc6-a7c6-4583-9a8d-d0e6809c3935.png)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2c30373b-3c44-c8bc-75d1-bb788efb479e
* Version Independent ID: 5ad284d8-3320-e7f3-0164-d29d3d82b74d
* Content: [Provision SharePoint assets from your SharePoint client-side web part](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/provision-sp-assets-from-package#update-an-existing-instance-in-the-site)
* Content Source: [docs/spfx/web-parts/get-started/provision-sp-assets-from-package.md](https://github.com/SharePoint/sp-dev-docs/blob/main/docs/spfx/web-parts/get-started/provision-sp-assets-from-package.md)
* Product: **sharepoint**
* Technology: **sharepoint-framework**
* GitHub Login: @VesaJuvonen
* Microsoft Alias: **vesaj**

**Comment by PooLP** [comment]
get it button appear only if component have a update.

**Comment by jared-leddy** [comment]
> get it button appear only if component have a update.

Under the "Added" button, it clearly states "There is a new version of this app. Get it now."

**Comment by jared-leddy** [comment]
> @jared-leddy - what level of permissions does your user have?

Level - Full Control, Owner Group
Site Collection Admin

**Comment by AJIXuMuK** [comment]
thanks for the confirmation @jared-leddy.
We're working on the issue.

**Comment by XinyangWang1024** [comment]
@jared-leddy Thanks for reporting this issue, our team is working on it. We may need some more time since we need to involve designer to give new design for this scenario.

Please try this **work around** before we fix this issue:

The URL of the page you see should be like: 
".../_layouts/15/appStore.aspx/appDetail/**2e22cf37-9c1b-403f-844e-5ff952ed684f**?sorting=1&from=0&catalog=1&id=2e22cf37-9c1b-403f-844e-5ff952ed684f"

Change the URL to below format and you could see the "get it" button:
".../_layouts/15/storefront.aspx#vw=AppDetailsView,app=**2e22cf37-9c1b-403f-844e-5ff952ed684f**,clg=1"

(This is the classic view of app detail page, usually we do not encourage users to use it. But for now, you could leverage this page to see the button)

**Comment by jared-leddy** [comment]
### Reply To:
> @jared-leddy Thanks for reporting this issue, our team is working on it... 
> Please try this **work around** before we fix this issue...

@XinyangWang1024 thanks for the info. That link specifically failed to work. I did, however, do the following which returned the below screenshot. 

### Steps to Complete
- Navigate to site contents
- Switch to classic view
- On the app, click the ellipsis and open the about page

### Screenshot
<img width="915" alt="image" src="https://user-images.githubusercontent.com/57743183/182517759-ea76dace-efe3-4645-b3ce-27378fa71682.png">

### Newly Designed Page
We would like to keep this ticket open until the newly designed update is completed. Our team of roughly 40 developers is very interested in getting this resolved.

**Comment by XinyangWang1024** [comment]
Hi @jared-leddy , yeah we can keep this ticket open. And we will keep you updated once we have some progress.

**Comment by XinyangWang1024** [comment]
Hi @jared-leddy , sorry for the late reply, my colleague has helped fix this issue. Please check the modern app store. Thanks!

**Comment by jared-leddy** [comment]
> Hi @jared-leddy , sorry for the late reply, my colleague has helped fix this issue. Please check the modern app store. Thanks!
Hey @XinyangWang1024,
Just tested this on our tenant. We did not see the updated modern page show the button to update the app.

**Comment by XinyangWang1024** [comment]
Hi @jared-leddy , we will display a "upgrade" button in the following case:

1. An **old version** is **feature deployed** in tenant app catalog, and is **added to a site**
2. A **new version** is uploaded to tenant app catalog and **feature deployed again**, then in the site where we once added this app, enter the app detail page, we will see a "upgrade" button
![image](https://user-images.githubusercontent.com/37166032/198683716-78e41c64-6826-4808-aaf4-f0f07ce5c52a.png)

**Comment by jared-leddy** [comment]
Yes. We need the upgrade button in the modern page. Right now, the only way to access this feature is to switch into classic mode. Which will eventually be discontinued.

**Comment by XinyangWang1024** [comment]
Hi @jared-leddy ,  we do have upgrade button in the modern page now. And upgrade button will show up in the following case:

1. Upload app to app catalog and **feature deploy**
![image](https://user-images.githubusercontent.com/37166032/199933332-1867a6e9-571b-4098-9d08-fc657ce49408.png)

2. Add the app to a site

3. Upload app's new version to app catalog and **feature deploy again**

4. in the site where the app is added, on "**my apps**" page, find the app in "**added apps**" section, click to enter **app detail page**, you will see the **upgrade** button.
![image](https://user-images.githubusercontent.com/37166032/199935615-2a021d9b-588f-4761-a303-9b456593a5e5.png)

![image](https://user-images.githubusercontent.com/37166032/199935667-a1ae7604-349d-45fd-b1ac-8267eae745a3.png)


Please follow the above steps to see if you can see the **Upgrade** button. Thanks! If you still cannot see the upgrade button, could you please record a video of your operation?

**Comment by jared-leddy** [comment]
> Hi @jared-leddy , we do have upgrade button in the modern page now. And upgrade button will show up in the following case...
>
Our `package-solution.json` file shows the following:
```json
"skipFeatureDeployment": true,
```
If we change this to false, the ability to see our app in the "app home" (under the add an app page) disappears. We have to set this to true in order to see our apps here. 

Will that be a problem?

**Comment by XinyangWang1024** [comment]
Hi @jared-leddy , the "**skipFeatureDeployment**" value works in below way:
1. if this value **is set to true**, then you have **2 choices** when you deploy the app in tenant app catalog
![image](https://user-images.githubusercontent.com/37166032/200246783-ca364d28-c6f4-4e58-936f-9ecd49a51318.png)
1.1 if you choose **first choice (feature deploy)**, you will need to manually add the app to the site where you want to use the app
1.2 if you choose the **second choice (tenant deploy)**, the app is available across all sites in your tenant, you do not need to manually add the app to a site.

2. if this value **is set to false**, then you only have **1 choice** when you deploy the app in tenant app catalog
![image](https://user-images.githubusercontent.com/37166032/200247172-c0cf593c-0b02-4796-b632-e344c1ed31ac.png)
This choice is **the same as 1.1**, you will need to manually add the app to the site where you want to use the app

I do not think this value is related to your issue. Could you please share the url of this page you mentioned?
`in the "app home" (under the add an app page)`

And please follow exactly the same steps as mentioned in the last comment, and see if you can see the upgrade button. (if you choose tenant deploy, then you do not need "upgrade button", because all sites in your tenant will use the latest version of app). Please let me know if you still cannot see the button

**Comment by jared-leddy** [comment]
Deployed as true. 
Can't provide URL to the "add an app" page. Org policy. 
Did not deploy to "all sites in the tenant". Only enable this app. 
Updated the app using the same config settings. 

The reason it's not deployed tenant wide is because multiple orgs are under 1 tenant. So, we have to manage the apps per site. 

The button is not present. I will have to switch to classic mode to view it. 
![image](https://user-images.githubusercontent.com/57743183/200353860-9ad8676f-dd9e-42e0-b138-b19903a684f2.png)

**Comment by XinyangWang1024** [comment]
Hi @jared-leddy ,thanks for the update. That's quite strange, since the fix works in our test tenants and should reach prod for a long time. Could you please help:

1.Check the **build version** (by right-click on the page, and click "view page source", then search for "buildNumber")
![image](https://user-images.githubusercontent.com/37166032/200484083-3281f658-41b1-4c9b-8e04-709d4c6a474e.png)

2. On the app detail page, could you please check the network, and search for "**appDetail**", and **take a screenshot** for the response? Because the display of button relies on the API response, analyzing the response may help us to further investigate
![image](https://user-images.githubusercontent.com/37166032/200484934-e8740f50-449c-41cc-815c-857ddfa400c5.png)

3. Does your tenant use any **site collection app catalog** besides **tenant app catalog**?

Thank you for the patience.

**Comment by jared-leddy** [comment]
Not completed.
